src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -449,6 +466,50 @@ private EppoValue getSubjectVariationOverride(
         return experimentConfiguration.getTypedOverrides().getOrDefault(hexedSubjectKey, new EppoValue());
     }
 
+    /***
+     * Logs an action taken that was not selected by the bandit.
+     * Useful for full transparency on what users experienced.
+     * @param subjectKey subjectKey identifier of the experiment subject, for example a user ID.
+     * @param flagKey feature flag, bandit, or experiment identifier
+     * @param subjectAttributes optional attributes associated with the subject, for example name, email, account age, etc.
+     * @param actionString name of the action taken for the subject
+     * @param actionAttributes attributes associated with the given action
+     * @return null if no exception was encountered by logging; otherwise, the encountered exception
+     */
+    public Exception logNonBanditAction(
+        String subjectKey,
+        String flagKey,
+        EppoAttributes subjectAttributes,
+        String actionString,
+        EppoAttributes actionAttributes
+    ) {
+        Exception loggingException = null;
+        try {
+            VariationAssignmentResult assignmentResult = this.getAssignedVariation(flagKey, subjectKey, subjectAttributes);
+
+            if (assignmentResult == null) {
+                // No bandit at play
+                return null;
+            }
+
+            String variationValue = assignmentResult.getVariation().getTypedValue().stringValue();
```

Comment:
```suggestion
            String variationValue = assignmentResult.getVariation().getTypedValue().toString()
```

`stringValue` assumes a "string` variation type right? Perhaps you meant to call `toString` on this to handle any type?

Reply:
This is a good call out! For now, we'll have all bandit experiments use string types only. But I can change to use `toString()` to help set us well for a future where that may not be the case.